pkg/rpc/server/listen.go

```diff
@@ -47,39 +65,29 @@ func ListenAndServe(c config.Config, fsCache cache.FSCache) error {
 		})
 	}
 
-	go func() {
-		worker := initializeDBWorker(c.CacheDir, true)
-		ctx := context.Background()
-		for {
-			time.Sleep(1 * time.Hour)
-			if err := worker.update(ctx, c.AppVersion, c.CacheDir, dbUpdateWg, requestWg); err != nil {
-				log.Logger.Errorf("%+v\n", err)
-			}
-		}
-	}()
-
 	mux := http.NewServeMux()
 
 	scanHandler := rpcScanner.NewScannerServer(initializeScanServer(fsCache), nil)
-	mux.Handle(rpcScanner.ScannerPathPrefix, withToken(withWaitGroup(scanHandler), c.Token, c.TokenHeader))
+	mux.Handle(rpcScanner.ScannerPathPrefix, withToken(withWaitGroup(scanHandler), token, tokenHeader))
 
 	layerHandler := rpcCache.NewCacheServer(NewCacheServer(fsCache), nil)
-	mux.Handle(rpcCache.CachePathPrefix, withToken(withWaitGroup(layerHandler), c.Token, c.TokenHeader))
+	mux.Handle(rpcCache.CachePathPrefix, withToken(withWaitGroup(layerHandler), token, tokenHeader))
 
 	// osHandler is for backward compatibility
 	osHandler := rpcDetector.NewOSDetectorServer(initializeOspkgServer(), nil)
-	mux.Handle(rpcDetector.OSDetectorPathPrefix, withToken(withWaitGroup(osHandler), c.Token, c.TokenHeader))
+	mux.Handle(rpcDetector.OSDetectorPathPrefix, withToken(withWaitGroup(osHandler), token, tokenHeader))
 
 	// libHandler is for backward compatibility
 	libHandler := rpcDetector.NewLibDetectorServer(initializeLibServer(), nil)
-	mux.Handle(rpcDetector.LibDetectorPathPrefix, withToken(withWaitGroup(libHandler), c.Token, c.TokenHeader))
+	mux.Handle(rpcDetector.LibDetectorPathPrefix, withToken(withWaitGroup(libHandler), token, tokenHeader))
 
 	mux.HandleFunc("/healthz", func(rw http.ResponseWriter, r *http.Request) {
-		rw.Write([]byte("ok"))
+		if _, err := rw.Write([]byte("ok")); err != nil {
+			log.Logger.Errorf("health check error: %s", err)
```

Comment:
Should we also send back an http status 200?

Reply:
Do you mean we should set status as follows?

```suggestion
		w.WriteHeader(http.StatusOK)
		if _, err := rw.Write([]byte("ok")); err != nil {
			log.Logger.Errorf("health check error: %s", err)
```

`Write` sets StatusOK automatically. Do you think we should specify it explicitly?
https://github.com/golang/go/blob/master/src/net/http/server.go#L1594-L1596

Or, are you talking about inside the if statement? If so, the error from `rw.Write` means that there is an issue on the HTTP connection and we can't return 200 in that case.

Reply:
Oh actually I didn't know that it sets it! That's cool! In that case, disregard my comment.